src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -60,19 +57,49 @@ typedef struct {
 static xfer_ctl_t xfer_status[DWC2_EP_MAX][2];
 #define XFER_CTL_BASE(_ep, _dir) (&xfer_status[_ep][_dir])
 
-// EP0 transfers are limited to 1 packet - larger sizes has to be split
-static uint16_t ep0_pending[2];  // Index determines direction as tusb_dir_t type
-static uint16_t _dfifo_top;      // top free location in DFIFO in words
+typedef struct {
+  CFG_TUD_MEM_ALIGN union {
+    uint32_t setup_packet[2];
+#if CFG_TUD_MEM_DCACHE_ENABLE
+    uint8_t setup_packet_cache_padding[CFG_TUD_MEM_DCACHE_LINE_SIZE];
```

Comment:
@roma-jam we will need to do this to all buffered in the class driver, which may take a couple of follow-up PRs.